app/src/main/java/org/simple/clinic/bp/entry/BloodPressureEntryEffectHandler.kt

```diff
@@ -1,12 +1,32 @@
 package org.simple.clinic.bp.entry
 
-import io.reactivex.Observable
+import com.spotify.mobius.rx2.RxMobius
 import io.reactivex.ObservableTransformer
+import org.simple.clinic.util.UserInputDatePaddingCharacter
+import org.simple.clinic.util.scheduler.SchedulersProvider
+import org.threeten.bp.LocalDate
 
 object BloodPressureEntryEffectHandler {
-  fun create(): ObservableTransformer<BloodPressureEntryEffect, BloodPressureEntryEvent> {
-    return ObservableTransformer {
-      Observable.never<BloodPressureEntryEvent>()
-    }
+  fun create(
+      ui: BloodPressureEntryUi,
+      inputDatePaddingCharacter: UserInputDatePaddingCharacter,
+      schedulersProvider: SchedulersProvider
+  ): ObservableTransformer<BloodPressureEntryEffect, BloodPressureEntryEvent> {
+    return RxMobius
+        .subtypeEffectHandler<BloodPressureEntryEffect, BloodPressureEntryEvent>()
+        .addConsumer(PrefillDate::class.java, { prefillDate(ui, it.date, inputDatePaddingCharacter) }, schedulersProvider.ui())
+        .build()
+  }
+
+  fun prefillDate(
```

Comment:
Ideally, the date to be displayed should be part of the model. I am assuming that we will change this once the feature is migrated?

Reply:
Yes, that is correct. We will do this once we starting moving the update patient entry flow to Mobius.